static/js/redux/ui/advising.jsx

```diff
@@ -73,6 +76,15 @@ class Advising extends React.Component {
   }
 
   componentDidMount() {
+    this.fetchSemesters();
+  }
+
+  fetchSemesters() {
+    fetch(getRetrievedSemesters())
+      .then(response => response.json())
+      .then((data) => {
+        this.setState({ displayed_semesters: this.state.displayed_semesters.concat(data.retrievedSemesters) })
```

Comment:
I wouldn't concatenate - instead just replace it.

Reply:
I think it's necessary to concatenate because the latest semester is not obtained from SIS, it's from the original dashboard.

Reply:
right - but we have a prop for semester that we can use i believe. We don't have the setState() for displayed_semesters to depend on the previous state of displayed_semesters if that makes sense

Reply:
above this line, you can do something like 
```let semesters = [this.props.semester]; //the current semester 
setState(displayed_semesters: semesters.concat(data.retrievedSemesters));```

Reply:
ugh my computer is glitching hold on

Reply:
not sure if the array initialization syntax is correct but u get the point